src/logger.cpp

```diff
@@ -1,18 +1,37 @@
 #include "logger.hpp"
+#include <boost/log/attributes/constant.hpp>
 
 namespace attrs   = boost::log::attributes;
 namespace expr    = boost::log::expressions;
 namespace logging = boost::log;
 
 // Defines a global logger initialization routine
-BOOST_LOG_GLOBAL_LOGGER_INIT(my_logger, src::severity_logger_mt<boost::log::trivial::severity_level>)
+BOOST_LOG_GLOBAL_LOGGER_INIT(my_logger, src::severity_logger_mt<logging::trivial::severity_level>)
 {
-  src::severity_logger_mt<boost::log::trivial::severity_level> lg;
+  src::severity_logger_mt<logging::trivial::severity_level> lg;
+  lg.add_attribute("ASTE", attrs::constant<bool>(true));
   logging::add_common_attributes();
-  auto format = expr::stream << "---[ASTE] " << expr::message;
-  logging::add_console_log(std::clog, keywords::format = format);
-  logging::core::get()->set_filter(
-      logging::trivial::severity >= logging::trivial::info);
-
   return lg;
 }
+
+void addLogIdentity(const std::string &participant, int rank)
+{
+  my_logger::get().add_attribute("Participant", attrs::constant<std::string>(participant));
+  my_logger::get().add_attribute("Rank", attrs::constant<int>(rank));
+}
+
+void addLogSink(bool verbose)
+{
+  std::string filter = "( %Severity% >= ";
+  filter.append(verbose ? "debug" : "info");
+  filter.append(" ) and %ASTE%");
+  std::cerr << filter;
```

Comment:
I don't get why we put the filter setting on the `cerr` stream?

Reply:
Thanks for pointing that out, this still was from debugging.